### PR TITLE
Jokeen/2022w23

### LINKED ISSUE #1803: AVR-915 `eadv` and `dis` appears to ignore `eadv` instead of canceling
**What is the bug?**: `eadv` and `dis` appears to ignore `eadv` instead of canceling. 
**Severity**: Medium (minor functional impact)
**Steps to reproduce**: Make an attack with `eadv` and `dis` arguments, or add an effect with `dis` and try to make an attack adding `eadv`.  Notice there are two d20 rolled, with disadvantage.
**Context**: Making an attack.

- TheReverendB#1377

### PR DESCRIPTION
**Pull request by Croebh**
### Summary
Implements AFR-953 - Adds `ctx.message_id` to Draconic to give access to the ID of the issuing command message.
Fixes #1803 - Fixes an edge case where the user had `dis` and `eadv` args, where it would apply disadvantage.
Adds unit tests for reconcile_adv()

#### PR Type

- [x] This PR is a code change that implements a feature request.
- [x] This PR fixes an issue.
- [ ] This PR adds a new feature that is not an open feature request.
- [ ] This PR is not a code change (e.g. documentation, README, ...)
#### Other
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [X] If code changes were made then they have been tested.
- [x] I have updated the documentation to reflect the changes.

### CODE REVIEW

File: aliasing/api/context.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def __init__(self, ctx):
         self._author = AliasAuthor(ctx.author)
         self._prefix = ctx.prefix
         self._alias = ctx.invoked_with
+        self._message = ctx.message.id
```
Comment (zhudotexe) [inline aliasing/api/context.py:21]: Let's name this attr `_message_id` if it's just saving the ID - the name should make it clear what lives there, and naming it just `_message` makes it seem like it saves the entire Message object.

Same for the public `.message` -> `.message_id` attr.

File: utils/functions.py [hunk 2]
```diff
@@ -490,7 +490,7 @@ def reconcile_adv(adv=False, dis=False, eadv=False) -> enums.AdvantageType:
     :return: The combined advantage result
     """
     result = 0
-    if adv:
+    if adv or eadv:
```
Comment (zhudotexe) [inline utils/functions.py:493]: Kind of not actually related to the content of the PR here, but can you add a unit test for this similar to the one in test_argparse for the 8 possible inputs/outputs?

File: tests/utils.py [hunk 3]
```diff
@@ -126,3 +127,8 @@ def guild(self):
     @property
     def author(self):
         return self.guild.get_member(int(DEFAULT_USER_ID))
+
+
+class MessageProxy:
+    def __init__(self):
+        self.id = 112358132134
```
[the remark this answers is not in the document]
Comment (Croebh) [reply tests/utils.py]: But I liked the fibbonaci sequence!
